pkg/ibm/ibmp.go

```diff
@@ -253,6 +261,67 @@ func (ibmp IBMPowerDynamicConfig) GetState(kubeClient client.Client, ctx context
 	return "OK", nil
 }
 
+// CleanUpVms deletes any VMs in the pw cloud instance that are associated with a non-existing Tekton TaskRun.
+// Each VM instance should have a tag with the associated TaskRun's namespace and name. This is compared to
+// existingTaskRuns, which is a map of namespaces to a list of TaskRuns in that namespace, to determine if this
+// VM's TaskRun still exists.
+func (pw IBMPowerDynamicConfig) CleanUpVms(kubeClient client.Client, ctx context.Context, existingTaskRuns map[string][]string) error {
+	// Get all VM instances
+	log := logr.FromContextOrDiscard(ctx)
+	log.Info("Attempting to clean up orphaned IBM Power Systems instances")
+	service, err := pw.createAuthenticatedBaseService(ctx, kubeClient)
+	if err != nil {
+		return fmt.Errorf("failed to create an authenticated base service: %w", err)
+	}
+	pvmInstancesCollection, err := pw.listInstances(ctx, service)
+	if err != nil {
+		return fmt.Errorf("failed to fetch Power Systems instances: %w", err)
+	}
+
+	// Iterate over all VM instances
+	for _, instanceRef := range pvmInstancesCollection.PvmInstances {
+		pvmInstance, err := pw.getInstance(ctx, service, *instanceRef.PvmInstanceID)
+		if err != nil {
+			msg := fmt.Sprintf("failed to get instance %s; skipping this instance...", *instanceRef.PvmInstanceID)
+			log.Error(err, msg)
```

Comment:
I'd suggest to use the KeyValue format
```suggestion
			log.Error("failed to get instance; skipping this instance...", "pvm-instance-id", *instanceRef.PvmInstanceID)
```

Reply:
Implemented!

Reply:
It seems to me it didn't get pushed